Find a functionally equivalent code:android.content.pm.ResolveInfo.getIconResource:COMMENT
Method Modifier: public      final       
Comment:/**
 * Return the icon resource identifier to use for this match.  If the
 * match defines an icon, that is used; else if the activity defines
 * an icon, that is used; else, the application icon is used.
 *
 * @return The icon associated with this match.
 */

@@ -1,8 +1,11 @@
 {
+    if (noResourceId)
+        return 0;
     if (icon != 0)
         return icon;
     final ComponentInfo ci = getComponentInfo();
-    if (ci != null)
+    if (ci != null) {
         return ci.getIconResource();
+    }
     return 0;
 }

